crates/tauri-runtime-wry/src/dialog/windows.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019-2024 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use windows::core::{w, HSTRING, PCWSTR};
+
+enum Level {
+  Error,
+  #[allow(unused)]
+  Warning,
+  #[allow(unused)]
+  Info,
+}
+
+pub fn error<S: AsRef<str>>(err: S) {
+  dialog_inner(err, Level::Error);
+}
+
+fn dialog_inner<S: AsRef<str>>(err: S, level: Level) {
+  let err = err.as_ref();
+
+  let title = match level {
+    Level::Warning => w!("Warning"),
+    Level::Error => w!("Error"),
+    Level::Info => w!("Info"),
+  };
+
+  #[cfg(not(feature = "common-controls-v6"))]
+  {
+    use windows::Win32::UI::WindowsAndMessaging::*;
+
+    let err = remove_hyperlink(err);
+    let err = HSTRING::from(err);
+    let err = PCWSTR(err.as_ptr());
+
+    unsafe {
+      MessageBoxW(
+        None,
+        err,
+        title,
+        match level {
+          Level::Warning => MB_ICONWARNING,
+          Level::Error => MB_ICONERROR,
+          Level::Info => MB_ICONINFORMATION,
+        },
+      )
+    };
+  }
+
+  #[cfg(feature = "common-controls-v6")]
+  {
+    use windows::core::HRESULT;
+    use windows::Win32::Foundation::*;
+    use windows::Win32::UI::Controls::*;
+    use windows::Win32::UI::Shell::*;
+    use windows::Win32::UI::WindowsAndMessaging::*;
+
+    extern "system" fn task_dialog_callback(
+      _hwnd: HWND,
+      msg: TASKDIALOG_NOTIFICATIONS,
+      _wparam: WPARAM,
+      lparam: LPARAM,
+      _data: isize,
+    ) -> HRESULT {
+      if msg == TDN_HYPERLINK_CLICKED {
+        let link = PCWSTR(lparam.0 as _);
+        let _ = unsafe { ShellExecuteW(None, None, link, None, None, SW_SHOWNORMAL) };
+      }
+
+      S_OK
+    }
+
+    let err = HSTRING::from(err);
+    let err = PCWSTR(err.as_ptr());
```

Comment:
```suggestion
```

Reply:
but this one can't because it is used in a struct with an owned value not reference

Reply:
Right right, I'm so paranoid nowadays when I see `PCWSTR` 😂